### ISSUE TITLE
GeolocationPositionOptions extern declaration necessitates cast

### ISSUE THREAD

**Issue by chrismiddleton** [google/closure-compiler#2060]
Right now the GeolocationPositionOptions extern is defined as a `@constructor`, which seems to mean that when calling `getCurrentPosition`, you have to cast to `GeolocationPositionOptions`. If it were defined as an `@interface` instead, would that allow a vanilla object to work?

MDN Reference:
https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition

Closure Compiler Externs File:
https://github.com/google/closure-compiler/blob/a0b884082dbef77d2d336705b32d99e0a930dc54/externs/browser/w3c_geolocation.js

Example taken from the reference above:
```js
var options = {
  enableHighAccuracy: true,
  timeout: 5000,
  maximumAge: 0
};

function success(pos) { /* ... */ }

function error(err) { /* ... */ }

navigator.geolocation.getCurrentPosition(success, error, options);
                                                         ^ gives a warning
```
The warning looks like this:

WARNING - actual parameter 3 of Geolocation.prototype.watchPosition does not match formal parameter
found   : {
  enableHighAccuracy: boolean, 
  maximumAge: number, 
  timeout: number
}
required: (GeolocationPositionOptions|null|undefined)
		{enableHighAccuracy:true, timeout:600000, maximumAge:0}

**Comment by ChadKillingsworth** [comment]
Yeah that's very old style (I think I was the original author). I believe the correct option now is for it to be an `@record` so there are no casts whatsoever.

**Comment by Dominator008** [comment]
@chrismiddleton Would you like to submit a pull request?

**Comment by chrismiddleton** [comment]
> Would you like to submit a pull request?

Just submitted one, thanks for the suggestion. And thanks, Chad, for pointing out the correct use of `@record`.

**Comment by ChadKillingsworth** [comment]
Closed by 800a9701170f848422e9fe020ed116d8cfaf2776
